src/main/scala/is/hail/rvd/RVDContext.scala

```diff
@@ -1,20 +1,58 @@
 package is.hail.rvd
 
-import is.hail.annotations.Region
+import is.hail.annotations.{Region, RegionValueBuilder}
+import is.hail.sparkextras.ResettableContext
+
+import scala.collection.mutable
 
 object RVDContext {
   def default: RVDContext = new RVDContext(Region())
 
   def fromRegion(region: Region): RVDContext = new RVDContext(region)
 }
 
-// NB: must be *Auto*Closeable because calling close twice is undefined behavior
-// (see AutoCloseable javadoc)
-class RVDContext(r: Region) extends AutoCloseable {
+class RVDContext(r: Region) extends ResettableContext {
+  private[this] val children: mutable.ArrayBuffer[AutoCloseable] = new mutable.ArrayBuffer()
+
+  private[this] def own(child: AutoCloseable): Unit = children += child
+
+  own(r)
+
+  def freshContext: RVDContext = {
+    val ctx = RVDContext.default
+    own(ctx)
+    ctx
+  }
+
   def region: Region = r // lifetime: element
 
   def partitionRegion: Region = r // lifetime: partition
 
+  private[this] val theRvb = new RegionValueBuilder(r)
+  def rvb = theRvb
+
+  def reset(): Unit = {
+    r.clear()
```

Comment:
So I kinda don't like this.  Won't the caller always have the region?  Why not just call clear on it?  The code will be way more readable that way.

Reply:
I think I misunderstand something. The caller certainly always has the region (it's just `RVDContext.region`) and they could certainly do `context.region.clear()`. I don't have strong feelings, but I guess the whole point of this PR is to move the notion of being resettable to the Context.

I've been thinking: we have a single context for a whole partition rather than a single context for a single element of a partition because memory allocation is expensive. If you accept that, then `reset` is the mechanism by which we get a "new" context for each element.